readme.md

```diff
@@ -1,123 +1,131 @@
+
+Creating test files is a two step process. The generation of edit test files requires a clean data file to be present.[The steps above](https://github.com/cfpb/hmda-test-files/tree/master/readme.md/#generating-clean-files) outline the process to create the clean data files. 
+
+Running the generate test files script modifies the data from the clean file and saves it with the name of the edit check that it will fail. If edit files already exist with the same names as those produced by this script, they will be overwrriten. These files are written to the appropriate directory based on edit type, such as 2019/edits_files/test_files/{Bank Name}/syntax/s300.txt. The file name indicates which edit the file is designed to test. If an edit has multiple conditions, a file will be made for each condition in the format 2019/edits_files/test_files/{Bank Name}/syntax/s301_1.txt.
+
+For 2019: 
+1. Navigate to the 2019/python directory.
+2. Run `python3 generate_2019_error_files.py`
+3. The error files for testing syntax, validity, and quality edit test files will be created in the following diretories:
+	- Syntax: `2019/edits_files/test_files/{Bank Name}/syntax`
+	- Validity: `2019/edits_files/test_files/{Bank Name}/validity`
+	- Quality: `2019/edits_files/test_files/{Bank Name}/quality`
+
+Note: the filepath and filename for test files can be adjusted in the [2019 Test Filepaths](https://github.com/cfpb/hmda-test-files/tree/master/2019/configurations/test_filepaths.yaml) YAML configuration file under the keys, `syntax_filepath`, `validity_filepath`, and `quality_filepath`.  
+
+For 2018: 
+1. Navigate to the 2018/python directory.
+2. Run `python3 generate_2018_error_files.py`
+3. The error files for testing syntax, validity, and quality edit test files will be created in the following diretories:
+	- Syntax: `2018/edits_files/test_files/{Bank Name}/syntax`
+	- Validity: `2018/edits_files/test_files/{Bank Name}/validity`
+	- Quality: `2018/edits_files/test_files/{Bank Name}/quality`
+
+Note: the filepath and filename for test files can be adjusted in the [2018 Test Filepaths](https://github.com/cfpb/hmda-test-files/tree/master/2018/configurations/test_filepaths.yaml) YAML configuration file under the keys, `syntax_filepath`, `validity_filepath`, and `quality_filepath`. 
+
+## Generating Large Files 
+Large files are used during load testing. Large test files are created using a clean test file base and the user-specified row count in each year's large file configuration:
+ [2019 Large File Configuration](https://github.com/cfpb/hmda-test-files/tree/master/2019/configurations/test_filepaths.yaml).
+ [2018 Large File Configuration](https://github.com/cfpb/hmda-test-files/tree/master/2018/configurations/test_filepaths.yaml).
+
+Configuration options include (with defaulted values): 
+* `source_filepath`: #Specify source filepath.  
```

Comment:
I like the differentiation marker for optional vs required configuration parameters.

Reply:
Sure!